frontend/components/bridge/BridgingSteps.tsx

```diff
@@ -1,29 +1,224 @@
+      Don&apos;t worry, your funds remain safe. You can access them by importing
+      your Pearl seed phrase into a compatible wallet, like MetaMask or
+      Coinbase.
+    </Text>
+
+    <Text className="text-sm text-lighter">
+      Ask for help in{' '}
+      <a href={SUPPORT_URL} target="_blank" rel="noopener noreferrer">
+        the Olas community Discord server {UNICODE_SYMBOLS.EXTERNAL_LINK}
+      </a>
+    </Text>
+  </Flex>
+);
+
+type Step = {
+  title: string;
+  status: Status;
+  subSteps: {
+    description: Nullable<string>;
+    txnLink: Maybe<string>;
+    isFailed?: boolean;
+    onRetry?: () => void;
+  }[];
+};
+
+const generateBridgeStep = (
+  status: Status,
+  subSteps: StepEvent[],
+): Omit<Step, 'title'> => {
+  return {
+    status,
+    subSteps: subSteps.map(({ symbol, status, txnLink }) => {
+      const isFailed = status === 'error';
+      const description = (() => {
+        if (status === 'finish') {
+          return `Bridging ${symbol} transaction complete.`;
+        }
+        if (status === 'error') {
+          return `Bridging ${symbol} failed.`;
+        }
+        return `Sending transaction...`;
+      })();
+
+      return { description, txnLink, isFailed };
+    }),
+  };
+};
+
+const generateMasterSafeCreationStep = (
+  status: Status,
+  txnLink: Maybe<string>,
+): Step => {
+  const isFailed = status === 'error';
+  const description = (() => {
+    if (status === 'finish') return 'Transaction complete.';
+    if (status === 'error') return 'Transaction failed.';
+    if (status === 'process') return 'Sending transaction...';
+    return null;
+  })();
+
+  return {
+    title: 'Create Master Safe',
+    status,
+    subSteps: [{ description, txnLink, isFailed }],
+  };
+};
+
+const generateMasterSafeTransferStep = (
+  status: Status,
+  subSteps: StepEvent[],
+): Step => {
+  return {
+    title: 'Transfer funds to the Master Safe',
+    status,
+    subSteps: subSteps.map(({ symbol, status, txnLink }) => {
+      const isFailed = status === 'error';
+      const description = (() => {
+        if (status === 'finish') {
+          return `Transfer ${symbol} transaction complete.`;
+        }
+        if (status === 'process') {
+          return 'Sending transaction...';
+        }
+        if (status === 'error') {
+          return `Transfer ${symbol} transaction failed.`;
+        }
+        return null;
+      })();
+
+      return { description, txnLink, isFailed };
+    }),
+  };
+};
+
+type StepEvent = {
+  symbol: TokenSymbol;
+  status: Status;
+  txnLink?: Maybe<string>;
+  onRetry?: () => void;
+};
+
+// TODO: simplify status
+type BridgingStepsProps = {
+  chainName: string;
+  bridge: {
+    status: Status;
+    executions: StepEvent[];
+  };
+  masterSafeCreation?: {
+    status: Status;
+    txnLink?: Maybe<string>;
+    onRetry?: () => void;
+  };
+  masterSafeTransfer?: {
+    status: Status;
+    transfers: StepEvent[];
+  };
+};
```

Comment:
wonder if we could go with a more general approach e.g.

```suggestion
type StepEvent = {
  text: string;
  status: Status;
  txnLink?: Maybe<string>;
  onRetry?: () => void;
};

// TODO: simplify status
type BridgingStepsProps = {
  chainName: string;
  steps: {
    status: Status;
    subSteps: StepEvent[];
  };
};
```

so that e.g. for master safe creation there's one substep with text, status and txnLink